### PR TITLE
Configuration enhancements

### LINKED ISSUE #10146: Configuration enhancements
Needs to be decomposed to multiple articles with headings and the way the configuration is read is confusing. It has some unnecessary things that need to be avoided. See this: https://stackoverflow.com/questions/53971432/configuration-getvalue-or-configuration-getsection-always-returns-null/53975950?noredirect=1#comment94791037_53975950

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0eb4d762-3910-2c57-8741-3306c15537b6
* Version Independent ID: ba2254a0-ea8e-9b28-e94b-edd0fc58c043
* Content: [Configuration in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/index?view=aspnetcore-2.2)
* Content Source: [aspnetcore/fundamentals/configuration/index.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/fundamentals/configuration/index.md)
* Product: **aspnet-core**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes #10146 

The only thing that I didn't address was ...

>  recovering values ... aren't distinguished very well from the sections that pertain to getting values into configuration in the first place

... because I don't think that splitting the content that way necessarily results in a better layout. This is very much organized **_by-provider_**, and I think that's probably ok.

Regarding splitting the whole topic into subtopics by-provider, there are topic maintenance problems with that approach: We're a small team, so we don't want to split this into eight new topics and samples (*Yikes!* lol) unless absolutely necessary (although management may eventually ask for that).

@Rick-Anderson This only deals with existing content, so we don't need engineering on this one.

### CODE REVIEW

File: aspnetcore/fundamentals/configuration/index.md [hunk 1]
```diff
@@ -1422,9 +1457,11 @@ The sample app calls `GetSection` with the `starship` key. The `starship` key-va
 
 ::: moniker-end
 
+`GetSection` is made available to an app by referencing the [Microsoft.Extensions.Configuration](https://www.nuget.org/packages/Microsoft.Extensions.Configuration/) package, which is made available to any app that references the [Microsoft.AspNetCore.App metapackage](xref:fundamentals/metapackage-app).
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/configuration/index.md]: `GetSection` is in the [Microsoft.Extensions.Configuration](https://www.nuget.org/packages/Microsoft.Extensions.Configuration/) package, which is included in the [Microsoft.AspNetCore.App metapackage](xref:fundamentals/metapackage-app).

File: aspnetcore/fundamentals/configuration/index.md [hunk 2]
```diff
@@ -110,7 +120,7 @@ When the file is read into configuration, unique keys are created to maintain th
 * section1:key0
 * section1:key1
 
-<xref:Microsoft.Extensions.Configuration.ConfigurationSection.GetSection*> and <xref:Microsoft.Extensions.Configuration.IConfiguration.GetChildren*> methods are available to isolate sections and children of a section in the configuration data. These methods are described later in [GetSection, GetChildren, and Exists](#getsection-getchildren-and-exists).
+<xref:Microsoft.Extensions.Configuration.ConfigurationSection.GetSection*> and <xref:Microsoft.Extensions.Configuration.IConfiguration.GetChildren*> methods are available to isolate sections and children of a section in the configuration data. These methods are described later in [GetSection, GetChildren, and Exists](#getsection-getchildren-and-exists). `GetSection` is made available to an app by referencing the [Microsoft.Extensions.Configuration](https://www.nuget.org/packages/Microsoft.Extensions.Configuration/) package, which is made available to any app that references the [Microsoft.AspNetCore.App metapackage](xref:fundamentals/metapackage-app).
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/configuration/index.md]: Global search/replace
is made available to an app by referencing the -> is in the
which is made available to any app that references the -> which is in the

Comment (Rick-Anderson) [inline aspnetcore/fundamentals/configuration/index.md]: `GetSection` is in the [Microsoft.Extensions.Configuration](https://www.nuget.org/packages/Microsoft.Extensions.Configuration/) package, which is in the [Microsoft.AspNetCore.App metapackage](xref:fundamentals/metapackage-app).

File: aspnetcore/fundamentals/configuration/index.md [hunk 3]
```diff
@@ -757,6 +768,8 @@ public class Program
 }
 ```
 
+The base path is set with <xref:Microsoft.Extensions.Configuration.FileConfigurationExtensions.SetBasePath*>. `SetBasePath` is made available to an app by referencing the [Microsoft.Extensions.Configuration.FileExtensions](https://www.nuget.org/packages/Microsoft.Extensions.Configuration.FileExtensions/) package, which is made available to any app that references the [Microsoft.AspNetCore.App metapackage](xref:fundamentals/metapackage-app).
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/configuration/index.md]: Do we really need to tell them it's in M.A.App? Guess it doesn't hurt.

Comment (guardrex) [reply aspnetcore/fundamentals/configuration/index.md]: Yeah, the reader could land into any of these sections from an internal or external bookmark. If we only say the package, they might add the package to an app that already references `App`.

**Pull request merged**